bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -0,0 +1,460 @@
+
+import static org.openhab.binding.tibber.internal.TibberBindingConstants.*;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigDecimal;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Properties;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketClose;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketConnect;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketError;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.tibber.internal.config.TibberConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link TibberHandler} is responsible for handling queries to/from Tibber API.
+ *
+ * @author Stian Kjoglum - Initial contribution
+ */
+@NonNullByDefault
+public class TibberHandler extends BaseThingHandler {
+    private static final int REQUEST_TIMEOUT = (int) TimeUnit.SECONDS.toMillis(20);
+    private final Logger logger = LoggerFactory.getLogger(TibberHandler.class);
+    private final Properties httpHeader = new Properties();
+    private final SslContextFactory sslContextFactory = new SslContextFactory(true);
+    private final Executor websocketExecutor = ThreadPoolManager.getPool("tibber.websocket");
+    private final TibberWebSocketListener socket = new TibberWebSocketListener();
+    private @NonNullByDefault({}) TibberConfiguration tibberConfig;
```

Comment:
There has been an agreement among reviewers to no longer allow `@NonNullByDefault({})` for configurations.
I think this is one of the accepted workarounds:
```suggestion
    private TibberConfiguration tibberConfig = new TibberConfiguration();
```